client/asset/eth/eth.go

```diff
@@ -751,8 +751,32 @@ func (*ExchangeWallet) PayFee(address string, regFee, feeRateSuggestion uint64)
 	return nil, asset.ErrNotImplemented
 }
 
-func (*ExchangeWallet) SwapConfirmations(ctx context.Context, id dex.Bytes, contract dex.Bytes, startTime time.Time) (confs uint32, spent bool, err error) {
-	return 0, false, asset.ErrNotImplemented
+// SwapConfirmations gets the number of confirmations and the spend status
+// for the specified swap.
+func (eth *ExchangeWallet) SwapConfirmations(ctx context.Context, _ dex.Bytes, secretHashB dex.Bytes,
+	_ time.Time, assetVersion uint32) (confs uint32, spent bool, err error) {
+
+	hdr, err := eth.node.bestHeader(ctx)
+	if err != nil {
+		return 0, false, fmt.Errorf("error fetching best header: %w", err)
+	}
+
+	var secretHash [32]byte
+	copy(secretHash[:], secretHashB)
+
+	swapData, err := eth.node.swap(ctx, secretHash, assetVersion)
+	if err != nil {
+		return 0, false, fmt.Errorf("error finding swap state: %w", err)
+	}
+
+	if swapData.State == dexeth.SSNone {
+		return 0, false, asset.CoinNotFoundError
+	}
+
+	spent = swapData.State >= dexeth.SSRedeemed
+	confs = uint32(hdr.Number.Uint64() - swapData.BlockHeight + 1)
```

Comment:
Looking good in this method, and it does seem to be alright if the `contract dex.Bytes` arg encoded both the asset version and the secret hash.   Open to that?

Reply:
i.e.

```go
func (eth *ExchangeWallet) SwapConfirmations(ctx context.Context, _ dex.Bytes, contractData dex.Bytes,
	_ time.Time) (confs uint32, spent bool, err error) {

	hdr, err := eth.node.bestHeader(ctx)
	if err != nil {
		return 0, false, fmt.Errorf("error fetching best header: %w", err)
	}

	if len(contractData) != 36 {
		return 0, false, fmt.Errorf("invalid contract data")
	}

	// contractData fully specifies this swap in terms of which contract version
	// is used and the secret hash that keys the unique swap.
	var secretHash [32]byte
	copy(secretHash[:], contractData[:32])
	assetVersion := binary.BigEndian.Uint32(contractData[33:])

	swapData, err := eth.node.swap(ctx, secretHash, assetVersion)
	if err != nil {
		return 0, false, fmt.Errorf("error finding swap state: %w", err)
	}

	if swapData.State == dexeth.SSNone {
		return 0, false, asset.CoinNotFoundError
	}

	spent = swapData.State >= dexeth.SSRedeemed
	confs = uint32(hdr.Number.Uint64() - swapData.BlockHeight + 1)

	return
}
```